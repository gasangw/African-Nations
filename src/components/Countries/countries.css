.grid-container {
  width: 100%;
  padding: 5px 3px;
  margin: -186px auto;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  justify-content: center;
}

@media only screen and (min-width: 600px) {
  .grid-container {
    grid-template-columns: 1fr 1fr 1fr;
    margin-top: 36px;
  }

  img.card-img-top {
    width: 100%;
    height: 7rem;
  }
}

@media only screen and (max-width: 600px) {
  .grid-container {
    grid-template-columns: 1fr 1fr;
    margin-top: 0;
  }

  .map.container {
    margin-top: 8px;
  }
}

@media only screen and (max-width: 375px) {
  .grid-container { grid-template-columns: 1fr; }
}
