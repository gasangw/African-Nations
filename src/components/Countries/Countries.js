import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import { getCountries } from '../../redux/countries/countries';
import Country from '../Country/Country';

const Countries = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries.countries);
  useEffect(() => {
    dispatch(getCountries());
  }, []);
  return (
    <>
      <Container className="map">
        <Row>
          <Col xs={6}>
            <div className="africa-image">
          <p>Look up your Favorite country and learn more about it</p>
            </div>
          </Col>
          <Col xs={6} className="righ-map-column" style={{ color: 'white', zIndex: '1' }}>
            <h3 style={{ paddingTop: '40px' }}>AFRICA POPULATION</h3>
          </Col>
        </Row>
      </Container>
      <Container className="grid-container">
        {Object.keys(countries).map((countryId) => {
          const {
            currencies, flags, name, population,
          } = countries[countryId];
          return (
            <Country
              key={population}
              currencies={currencies}
              flags={flags}
              name={name}
              population={population}
            />
          );
        })}
      </Container>
    </>
  );
};

export default Countries;
