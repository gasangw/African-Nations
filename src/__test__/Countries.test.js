import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Countries from '../components/Countries/Countries';
import store from '../redux/configureStore';

it('Countries renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <Countries />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
