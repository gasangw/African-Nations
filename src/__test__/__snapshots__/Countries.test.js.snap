// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders correctly 1`] = `
Array [
  <div
    className="map container"
  >
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <div
          className="africa-image"
        >
          <img
            alt="africa map"
            src="images.jpg"
          />
        </div>
      </div>
      <div
        className="righ-map-column col-6"
        style={
          Object {
            "color": "blue",
            "zIndex": "1",
          }
        }
      >
        <h3
          style={
            Object {
              "paddingTop": "35px",
            }
          }
        >
          AFRICAN COUNTRIES
        </h3>
      </div>
    </div>
  </div>,
  <div
    className="grid-container container"
  />,
]
`;
